src/common/outputService/deployOutputService.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable no-console, class-methods-use-this , no-case-declarations */
+
+import { Connection, Messages } from '@salesforce/core';
+import { ChangeBundle, ChangeBundleInstall, WorkItem, WorkItemPromote } from '../types';
+import { DeploySummaryQueryResult, selectDeployAORSummaryData } from '../selectors/deployProgressSummarySelector';
+import { selectOrgUrl } from '../selectors/endpointSelector';
+import { EnvQueryResult, selectPipelineStageByEnvironment } from '../selectors/environmentSelector';
+import { WorkItemsQueryResult, selectWorkItemsByChangeBundles } from '../selectors/workItemSelector';
+import { AsyncOperationType } from '../constants';
+import { OutputService } from './outputService';
+
+Messages.importMessagesDirectory(__dirname);
+const output = Messages.loadMessages('@salesforce/plugin-devops-center', 'deploy.output');
+
+/**
+ * This type is used for storing all the info needed for the deployment summary
+ */
+type DeploySummary = {
+  stageName: string;
+  // It is a map if it is a soup or versioned promote
+  // It is an array of strings if it is an ad hoc promote
+  workItems: Map<string, string[]> | string[];
+  branchName: string;
+  orgUrl: string;
+};
+
+/**
+ * Inner type to avoid duplicated code
+ */
+type VersionedOrSoupDeploySummary = {
+  stageName: string;
+  orgUrl: string;
+};
+
+/**
+ * Service class to print the deploy output
+ *
+ * @author JuanStenghele-sf
+ */
+export class DeployOutputService extends OutputService {
```

Comment:
See my comment below on the OutputService class hierarchy.  Besides that, I think this needs to be broken up.  This class should just deal with outputting information when asked.  A new class hierarchy something like

interface DeploySummaryProvider {
  DeploySummary createSummary();
}
class AdHocPromoteSummaryProvider implements SummaryProvider {}
class SoupPromoteSummaryProvider implements SummaryProvider {}
class VersionedPromoteSummaryProvider implements SummaryProvider {}